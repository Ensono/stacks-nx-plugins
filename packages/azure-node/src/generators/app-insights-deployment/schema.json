{
  "$schema": "http://json-schema.org/schema",
  "$id": "AppInsightsDeployment",
  "title": "",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "alias": "p",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What is the name of the project for this component?"
    },
    "applicationinsightsConnectionString": {
      "type": "string",
      "description": "The env variable that stores the app insights connection string.",
      "alias": "e",
      "default": "APPLICATIONINSIGHTS_CONNECTION_STRING",
      "x-prompt": "What is the env variable for the application insights connection string?",
      "pattern": "^[A-Z]+(?:_[A-Z]+)*$",
      "minLength": 5,
      "maxLength": 100
    }
  },
  "required": [
    "project",
    "applicationinsightsConnectionString"
  ]
}
