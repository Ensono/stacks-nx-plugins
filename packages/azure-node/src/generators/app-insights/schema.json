{
    "$schema": "http://json-schema.org/schema",
    "$id": "AppInsights",
    "title": "Ensono Stacks Azure Node AppInsights Generator",
    "description": "The @ensono-stacks/azure-node:app-insights generator will add and configure App Insights into a azure/node project",
    "type": "object",
    "properties": {
        "project": {
            "type": "string",
            "description": "The name of the project.",
            "alias": "p",
            "$default": {
                "$source": "projectName"
            },
            "x-prompt": "What is the name of the project for this component?"
        },
        "applicationinsightsConnectionString": {
            "type": "string",
            "description": "The env variable that stores the app insights connection string.",
            "alias": "e",
            "default": "APPLICATIONINSIGHTS_CONNECTION_STRING",
            "x-prompt": "What is the env variable for the application insights connection string?",
            "pattern": "^[A-Z]+(?:_[A-Z]+)*$",
            "minLength": 5,
            "maxLength": 100
        },
        "server": {
            "type": "string",
            "description": "The name of your custom server file",
            "alias": "s",
            "x-prompt": "What is the path within your app to the server?"
        }
    },
    "required": ["project", "applicationinsightsConnectionString", "server"]
}
