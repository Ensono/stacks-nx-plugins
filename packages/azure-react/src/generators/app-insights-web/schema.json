{
    "$schema": "http://json-schema.org/schema",
    "cli": "nx",
    "$id": "AppInsightsWeb",
    "title": "Ensono Stacks Azure React App Insights Web Generator",
    "description": "The @ensono-stacks/azure-react:app-insights-web generator will install and configures a ReactJs app using the NX/React plugin then install and configures App Insights.",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "Library name",
            "$default": {
                "$source": "argv",
                "index": 0
            },
            "x-prompt": "What name would you like to use for the library?",
            "pattern": "^[a-zA-Z].*$"
        },
        "connectionString": {
            "type": "string",
            "description": "The env variable for the connection string.",
            "x-prompt": "What is the env variable for the app insights connection string?"
        },
        "directory": {
            "type": "string",
            "description": "A directory where the lib is placed.",
            "alias": "dir"
        },
        "importPath": {
            "type": "string",
            "description": "What import path would you like to use for the library?"
        },
        "tags": {
            "type": "string",
            "description": "Add tags to the library (used for linting).",
            "alias": "t"
        }
    },
    "required": ["name", "connectionString"]
}
