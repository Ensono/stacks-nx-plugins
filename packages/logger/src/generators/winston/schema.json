{
    "$schema": "http://json-schema.org/schema",
    "cli": "nx",
    "$id": "WinstonLogger",
    "title": "Ensono Stacks Logger Winston Generator",
    "description": "The @ensono-stacks/logger:winston generator will generate a new Nx library which contains a Winston logger instance and associated config.",
    "type": "object",
    "properties": {
        "name": {
            "type": "string",
            "description": "",
            "$default": {
                "$source": "argv",
                "index": 0
            },
            "x-prompt": "What name would you like to use?",
            "pattern": "[a-z0-9]+(?:-[a-z0-9]+)*"
        },
        "tags": {
            "type": "string",
            "description": "Add tags to the project (used for linting)",
            "alias": "t"
        },
        "directory": {
            "type": "string",
            "description": "A directory where the project is placed"
        },
        "skipFormat": {
            "description": "Skip fomatting files.",
            "type": "boolean",
            "default": false
        },
        "logLevelType": {
            "description": "Add the type of log levels that will be used.",
            "type": "string",
            "enum": ["cli", "syslog", "npm"],
            "default": "npm",
            "x-prompt": "What type of logging levels would you like to use?"
        },
        "consoleLogs": {
            "description": "Output logs to the console.",
            "type": "boolean",
            "default": false
        },
        "fileTransportPath": {
            "description": "Add a filepath to transport logs to",
            "type": "string"
        },
        "httpTransportHost": {
            "description": "Remote host of the HTTP logging endpoint",
            "type": "string"
        },
        "httpTransportPort": {
            "description": "Remote port of the HTTP logging endpoint",
            "type": "number"
        },
        "httpTransportPath": {
            "description": "Remote URI of the HTTP logging endpoint",
            "type": "string"
        },
        "httpTransportSSL": {
            "description": "Value indicating if we should use HTTPS",
            "type": "boolean",
            "default": false
        },
        "streamPath": {
            "description": "Stream transport path",
            "type": "string"
        }
    },
    "required": ["name"]
}
