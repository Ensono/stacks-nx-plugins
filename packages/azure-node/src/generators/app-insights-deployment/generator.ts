import {
    executedDependantGenerator,
    hasGeneratorExecutedForProject,
    verifyPluginCanBeInstalled,
} from '@ensono-stacks/core';
import { readProjectConfiguration, Tree } from '@nrwl/devkit';

import { AppInsightsDeploymentGeneratorSchema } from './schema';
import { updateDockerfile } from './utils/update-dockerfile';
import { updatePipelineStagesYaml } from './utils/update-pipeline';
import { updateProjectJsonHelmUpgradeTarget } from './utils/update-targets';

// eslint-disable-next-line consistent-return
export default async function appInsightsDeploymentGenerator(
    tree: Tree,
    options: AppInsightsDeploymentGeneratorSchema,
) {
    verifyPluginCanBeInstalled(tree, options.project);

    if (
        !executedDependantGenerator(
            tree,
            'WorkspaceDeployment',
            options.project,
        )
    )
        return false;
    if (
        hasGeneratorExecutedForProject(
            tree,
            options.project,
            'AzureNodeAppInsightsDeployment',
        )
    )
        return false;

    const project = readProjectConfiguration(tree, options.project);

    // Update project.json
    updateProjectJsonHelmUpgradeTarget(project, tree, options);
    updatePipelineStagesYaml(tree, options);
    updateDockerfile(project, tree);
}
