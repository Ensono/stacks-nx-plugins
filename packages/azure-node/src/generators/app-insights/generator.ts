import {
    tsMorphTree,
    addIgnoreEntry,
    formatFiles,
    thirdPartyDependencyWarning,
    deploymentGeneratorMessage,
    hasGeneratorExecutedForProject,
    verifyPluginCanBeInstalled,
} from '@ensono-stacks/core';
import {
    addDependenciesToPackageJson,
    getProjects,
    joinPathFragments,
    Tree,
} from '@nx/devkit';
import { runTasksInSerial } from '@nx/workspace/src/utilities/run-tasks-in-serial';
import chalk from 'chalk';
import path from 'path';

import { AppInsightsGeneratorSchema } from './schema';
import {
    initAppInsights,
    configureAppInsights,
    startAppInsights,
} from './templates/appInsights';
import { appInsightsVersion } from '../../../utils/versions';

function updateDependencies(tree: Tree) {
    return addDependenciesToPackageJson(
        tree,
        {
            applicationinsights: appInsightsVersion,
        },
        {},
    );
}

export default async function appInsightsGenerator(
    tree: Tree,
    options: AppInsightsGeneratorSchema,
) {
    verifyPluginCanBeInstalled(tree, options.project);

    if (
        hasGeneratorExecutedForProject(
            tree,
            options.project,
            'AzureNodeAppInsights',
        )
    )
        return false;

    const { applicationinsightsConnectionString, server } = options;
    const project = getProjects(tree).get(options.project);

    if (!project) {
        throw new Error(
            `No application was found with the name '${options.project}'`,
        );
    }

    const customServerPath = path.join(project.sourceRoot, server);

    // Check if custom server exist
    if (!tree.exists(customServerPath)) {
        throw new Error('No custom server found.');
    }

    const morphTree = tsMorphTree(tree);

    // Read the Node from the source file
    const customServer = morphTree.addSourceFileAtPath(customServerPath);

    const isAppInsightsImport = customServer
        .getImportDeclarations()
        .some(
            importDeclaration =>
                importDeclaration.getModuleSpecifier().getLiteralValue() ===
                'applicationinsights',
        );

    // Add appInsights to import if it's not there
    if (isAppInsightsImport) {
        throw new Error('AppInsights SDK is already in use.');
    }

    thirdPartyDependencyWarning(['@nx/next:custom-server']);

    customServer.addImportDeclaration({
        namespaceImport: 'appInsights',
        moduleSpecifier: 'applicationinsights',
    });

    const mainFunction = customServer.getFunction('main');

    // Add appInisghts statements
    mainFunction.insertStatements(
        0,
        initAppInsights(applicationinsightsConnectionString),
    );
    mainFunction.insertStatements(1, configureAppInsights(project.name));
    mainFunction.insertStatements(2, startAppInsights());

    // Add empty lines after appInisghts statements
    mainFunction.getStatements().forEach(statement => {
        if (statement.getText().includes('appInsights')) {
            statement.appendWhitespace(writer => writer.newLine());
        }
    });

    // Save changes
    customServer.saveSync();

    const serverPath = joinPathFragments(project.root, server);
    // add nrwl/next custom server to prettier ignore
    addIgnoreEntry(tree, '.prettierignore', 'next server', [serverPath]);
    // Format files excluding the server file
    await formatFiles(tree, [serverPath]);

    console.warn(
        chalk.yellow`${serverPath} has been added to .prettierignore; Amend this file to resolve linting issues.`,
    );

    // Add dependencies and install
    return runTasksInSerial(updateDependencies(tree), () =>
        deploymentGeneratorMessage(
            tree,
            `nx g @ensono-stacks/azure-node:app-insights-deployment --project ${options.project} --applicationinsightsConnectionString ${applicationinsightsConnectionString}`,
        ),
    );
}
