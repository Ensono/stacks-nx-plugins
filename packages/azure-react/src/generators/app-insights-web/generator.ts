import {
    hasGeneratorExecutedForProject,
    NormalizedSchema,
    normalizeOptions,
    verifyPluginCanBeInstalled,
} from '@ensono-stacks/core';
import {
    addDependenciesToPackageJson,
    formatFiles,
    generateFiles,
    offsetFromRoot,
    readProjectConfiguration,
    Tree,
} from '@nx/devkit';
import { determineProjectNameAndRootOptions } from '@nx/devkit/src/generators/project-name-and-root-utils';
import { Linter } from '@nx/eslint';
import { libraryGenerator } from '@nx/react';
import path from 'path';

import { AppInsightsWebGeneratorSchema } from './schema';
import updateTsConfig from './utils/tsconfig';
import {
    appInsightsReactVersion,
    appInsightsWebVersion,
} from '../../../utils/versions';

function updateDependencies(tree: Tree) {
    return addDependenciesToPackageJson(
        tree,
        {
            '@microsoft/applicationinsights-web': appInsightsWebVersion,
            '@microsoft/applicationinsights-react-js': appInsightsReactVersion,
        },
        {},
    );
}

function addFiles(
    tree: Tree,
    options: NormalizedSchema<AppInsightsWebGeneratorSchema>,
) {
    const templateOptions = {
        ...options,
        applicationinsightsConnectionString:
            options.applicationinsightsConnectionString,
        offsetFromRoot: offsetFromRoot(options.directory),
        template: '',
    };
    generateFiles(
        tree,
        path.join(__dirname, 'files'),
        options.projectRoot,
        templateOptions,
    );
}

export default async function appInsightsWebGenerator(
    tree: Tree,
    options: AppInsightsWebGeneratorSchema,
) {
    verifyPluginCanBeInstalled(tree);

    if (
        hasGeneratorExecutedForProject(
            tree,
            options.name,
            'AzureReactAppInsightsWeb',
        )
    )
        return false;

    if (!options.applicationinsightsConnectionString) {
        throw new Error('applicationinsightsConnectionString cannot be empty.');
    }

    const normalizedOptions = await normalizeOptions(tree, options, 'library');

    // Use the existing library generator
    await libraryGenerator(tree, {
        ...options,
        linter: Linter.EsLint,
        style: 'none',
        minimal: true,
    });

    // Generate files
    addFiles(tree, normalizedOptions);

    const project = readProjectConfiguration(tree, options.name);
    updateTsConfig(tree, path.join(project.root, 'tsconfig.json'));

    // Format files
    await formatFiles(tree);

    // Update package.json and install dependencies
    return updateDependencies(tree);
}
