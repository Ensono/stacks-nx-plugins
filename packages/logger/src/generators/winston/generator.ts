import {
    formatFilesWithEslint,
    addCustomTestConfig,
    NormalizedSchema as BaseNormalizedSchema,
    normalizeOptions,
    verifyPluginCanBeInstalled,
} from '@ensono-stacks/core';
import {
    addDependenciesToPackageJson,
    formatFiles,
    generateFiles,
    GeneratorCallback,
    names,
    offsetFromRoot,
    readProjectConfiguration,
    Tree,
    runTasksInSerial,
} from '@nx/devkit';
import { libraryGenerator } from '@nx/js';
import path from 'path';

import { WinstonLoggerGeneratorSchema } from './schema';
import { addEslint } from './utils/eslint';
import { WINSTON_VERSION } from './utils/version';

type NormalizedSchema = BaseNormalizedSchema<WinstonLoggerGeneratorSchema>;

function updateDependencies(tree: Tree) {
    return addDependenciesToPackageJson(
        tree,
        {
            winston: WINSTON_VERSION,
        },
        {},
    );
}

function addFiles(tree: Tree, options: NormalizedSchema) {
    const templateOptions = {
        ...options,
        ...names(options.name),
        offsetFromRoot: offsetFromRoot(options.projectRoot),
        template: '',
    };

    generateFiles(
        tree,
        path.join(__dirname, 'files'),
        options.projectRoot,
        templateOptions,
    );
}

export default async function generate(
    tree: Tree,
    options: WinstonLoggerGeneratorSchema,
) {
    verifyPluginCanBeInstalled(tree);

    const tasks: GeneratorCallback[] = [];
    const normalizedOptions = await normalizeOptions(tree, options, 'library');

    await libraryGenerator(tree, options);
    // Delete the default generated lib folder
    tree.delete(path.join(normalizedOptions.projectRoot, 'src', 'lib'));

    // Generate files
    addFiles(tree, normalizedOptions);

    const project = readProjectConfiguration(
        tree,
        normalizedOptions.projectName,
    );

    tasks.push(
        updateDependencies(tree),
        addEslint(tree, project.root),
        formatFilesWithEslint(options.name),
    );

    const ciCoverageConfig = {
        ci: {
            collectCoverage: true,
            coverageReporters: ['text', 'html'],
            collectCoverageFrom: ['./**/*.{js,jsx,ts,tsx}', './!**/*.config.*'],
            codeCoverage: true,
            ci: true,
        },
    };

    if (project.name) {
        addCustomTestConfig(tree, project, ciCoverageConfig);
    }

    // Format files
    if (!options.skipFormat) {
        await formatFiles(tree);
    }

    // Update package.json
    return runTasksInSerial(...tasks);
}
