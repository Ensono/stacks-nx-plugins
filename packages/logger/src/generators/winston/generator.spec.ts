import { readJson, readProjectConfiguration, Tree } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';

import generator from './generator';
import { WinstonLoggerGeneratorSchema } from './schema';

describe('logger generator', () => {
    let tree: Tree;
    const options: WinstonLoggerGeneratorSchema = {
        name: 'test-client',
        logLevelType: 'npm',
        consoleLogs: false,
        fileTransportPath: undefined,
        httpTransportPath: undefined,
        httpTransportHost: undefined,
        httpTransportPort: undefined,
        httpTransportSSL: undefined,
        streamPath: undefined,
    };

    beforeEach(() => {
        tree = createTreeWithEmptyWorkspace();
    });

    it('should generate the logger', async () => {
        await generator(tree, {
            ...options,
            directory: 'custom',
            tags: 'test, client',
        });

        const config = readProjectConfiguration(tree, 'custom-test-client');

        expect(config).toBeDefined();
        expect(config.tags).toEqual(['test', 'client']);
        expect(tree.exists('test-client/src/index.ts'));
        expect(tree.exists('test-client/src/index.test.ts'));
    });

    it('should install winston as a dependency', async () => {
        await generator(tree, options);

        const packageJson = readJson(tree, 'package.json');
        expect(Object.keys(packageJson.dependencies)).toEqual(
            expect.arrayContaining(['winston']),
        );
    });

    it('should add console log transport', async () => {
        await generator(tree, { ...options, consoleLogs: true });
        const indexFile = tree.read('/test-client/src/index.ts', 'utf-8');
        expect(indexFile).toContain(
            'logger.add(new winston.transports.Console())',
        );
    });

    it('should add file transport', async () => {
        await generator(tree, {
            ...options,
            fileTransportPath: '/log-file.log',
        });
        const indexFile = tree.read('/test-client/src/index.ts', 'utf-8');
        expect(indexFile).toContain(
            'logger.add(new winston.transports.File({filename: "/log-file.log"}))',
        );
    });

    it('should add http transport', async () => {
        await generator(tree, {
            ...options,
            httpTransportPath: '/testPath',
            httpTransportPort: 300,
            httpTransportHost: 'www.testsite.co.uk',
        });
        const indexFile = tree.read('/test-client/src/index.ts', 'utf-8');
        expect(indexFile).toContain(
            'const httpTransportConfiguration: winston.transports.HttpTransportOptions = {',
        );
        expect(indexFile).toContain(`host: 'www.testsite.co.uk'`);
        expect(indexFile).toContain('port: 300');
        expect(indexFile).toContain(`path: '/testPath'`);
        expect(indexFile).toContain(
            'logger.add(new winston.transports.Http(httpTransportConfiguration));',
        );
    });
    it('should add stream transport', async () => {
        await generator(tree, {
            ...options,
            streamPath: 'www.testsite.co.uk',
        });
        const indexFile = tree.read('/test-client/src/index.ts', 'utf-8');
        expect(indexFile).toContain(
            'logger.add(new winston.transports.Stream(streamTransportConfiguration));',
        );
    });

    it('should add the ci config in the test command in the project.json', async () => {
        await generator(tree, options);

        const projectConfig = readJson(tree, 'test-client/project.json');

        expect(projectConfig.targets.test).toMatchObject(
            expect.objectContaining({
                configurations: {
                    ci: {
                        ci: true,
                        collectCoverage: true,
                        coverageReporters: ['text', 'html'],
                        collectCoverageFrom: [
                            './**/*.{js,jsx,ts,tsx}',
                            './!**/*.config.*',
                        ],
                        codeCoverage: true,
                    },
                },
            }),
        );
    });

    it('should add the ci config in the test command in the project.json with a custom directory', async () => {
        await generator(tree, { ...options, directory: 'custom' });

        const projectConfig = readJson(tree, 'custom/test-client/project.json');

        expect(projectConfig.targets.test).toMatchObject(
            expect.objectContaining({
                configurations: {
                    ci: {
                        ci: true,
                        collectCoverage: true,
                        coverageReporters: ['text', 'html'],
                        collectCoverageFrom: [
                            './**/*.{js,jsx,ts,tsx}',
                            './!**/*.config.*',
                        ],
                        codeCoverage: true,
                    },
                },
            }),
        );
    });
});
