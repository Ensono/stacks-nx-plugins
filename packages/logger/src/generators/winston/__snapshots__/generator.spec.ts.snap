// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger generator should generate the logger: .prettierignore 1`] = `
"# Add files here to ignore them from prettier formatting
/dist
/coverage
/.nx/cache
/.nx/workspace-data"
`;

exports[`logger generator should generate the logger: .prettierrc 1`] = `
"{ "singleQuote": true }
"
`;

exports[`logger generator should generate the logger: custom/test-logger/.eslintrc.json 1`] = `
"{
  "ignorePatterns": ["!**/*"],
  "env": {
    "jest": true
  },
  "extends": ["../../.eslintrc.json", "plugin:testing-library/react"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx", "*.js", "*.jsx"],
      "rules": {
        "@typescript-eslint/no-floating-promises": "error",
        "testing-library/await-async-utils": "error",
        "testing-library/await-async-queries": "error",
        "testing-library/no-wait-for-side-effects": "error",
        "testing-library/no-manual-cleanup": "error",
        "testing-library/prefer-explicit-assert": "warn",
        "testing-library/prefer-presence-queries": "warn",
        "testing-library/prefer-user-event": "warn",
        "testing-library/no-debug": "off"
      },
      "excludedFiles": ["jest.config.ts"],
      "parserOptions": {
        "project": ["tsconfig(.*)?.json"]
      }
    },
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {}
    },
    {
      "files": ["*.js", "*.jsx"],
      "rules": {}
    },
    {
      "files": ["*.json"],
      "parser": "jsonc-eslint-parser",
      "rules": {
        "@nx/dependency-checks": [
          "error",
          {
            "ignoredFiles": [
              "{projectRoot}/eslint.config.{js,cjs,mjs,ts,cts,mts}"
            ]
          }
        ]
      }
    },
    {
      "files": "jest.config.ts",
      "rules": {
        "unicorn/no-abusive-eslint-disable": "off"
      }
    }
  ]
}
"
`;

exports[`logger generator should generate the logger: custom/test-logger/jest.config.ts 1`] = `
"export default {
  displayName: 'test-logger',
  preset: '../../jest.preset.js',
  testEnvironment: 'node',
  transform: {
    '^.+\\\\.[tj]s$': ['ts-jest', { tsconfig: '<rootDir>/tsconfig.spec.json' }],
  },
  moduleFileExtensions: ['ts', 'js', 'html'],
  coverageDirectory: '../../coverage/custom/test-logger',
};
"
`;

exports[`logger generator should generate the logger: custom/test-logger/package.json 1`] = `
"{
  "name": "@proj/test-logger",
  "version": "0.0.1",
  "private": true,
  "type": "commonjs",
  "main": "./src/index.js",
  "types": "./src/index.d.ts",
  "dependencies": {
    "tslib": "^2.3.0"
  }
}
"
`;

exports[`logger generator should generate the logger: custom/test-logger/project.json 1`] = `
"{
  "name": "test-logger",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "custom/test-logger/src",
  "projectType": "library",
  "tags": ["test", "logger"],
  "targets": {
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/custom/test-logger",
        "main": "custom/test-logger/src/index.ts",
        "tsConfig": "custom/test-logger/tsconfig.lib.json",
        "assets": ["custom/test-logger/*.md"]
      }
    },
    "lint": {
      "executor": "@nx/eslint:lint"
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "custom/test-logger/jest.config.ts"
      },
      "configurations": {
        "ci": {
          "collectCoverage": true,
          "coverageReporters": ["text", "html"],
          "collectCoverageFrom": ["./**/*.{js,jsx,ts,tsx}", "./!**/*.config.*"],
          "codeCoverage": true,
          "ci": true
        }
      }
    }
  }
}
"
`;

exports[`logger generator should generate the logger: custom/test-logger/src/index.test.ts 1`] = `
"import logger from '.';

describe('logger', () => {
  it('should be a winston logger instance', () => {
    expect(logger).toBeDefined();

    expect(logger).toHaveProperty('error');
    expect(logger).toHaveProperty('warn');
    expect(logger).toHaveProperty('info');
    expect(logger).toHaveProperty('verbose');
    expect(logger).toHaveProperty('debug');
    expect(logger).toHaveProperty('silly');
  });
});
"
`;

exports[`logger generator should generate the logger: custom/test-logger/src/index.ts 1`] = `
"import * as winston from 'winston';

const logConfiguration: winston.LoggerOptions = {
  level: 'debug',
  levels: winston.config.npm.levels,
};

if (process.env.NODE_ENV === 'production') {
  logConfiguration.level = 'info';
}

// Reusable winston logging instance
const logger = winston.createLogger(logConfiguration);

export default logger;
"
`;

exports[`logger generator should generate the logger: custom/test-logger/tsconfig.json 1`] = `
"{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "module": "commonjs"
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    }
  ]
}
"
`;

exports[`logger generator should generate the logger: custom/test-logger/tsconfig.lib.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../dist/out-tsc",
    "declaration": true,
    "types": ["node"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["jest.config.ts", "src/**/*.spec.ts", "src/**/*.test.ts"]
}
"
`;

exports[`logger generator should generate the logger: custom/test-logger/tsconfig.spec.json 1`] = `
"{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "outDir": "../../dist/out-tsc",
    "module": "commonjs",
    "moduleResolution": "node10",
    "types": ["jest", "node"]
  },
  "include": [
    "jest.config.ts",
    "src/**/*.test.ts",
    "src/**/*.spec.ts",
    "src/**/*.d.ts"
  ]
}
"
`;

exports[`logger generator should generate the logger: jest.config.ts 1`] = `
"import type { Config } from 'jest';
import { getJestProjectsAsync } from '@nx/jest';

export default async (): Promise<Config> => ({
  projects: await getJestProjectsAsync(),
});
"
`;

exports[`logger generator should generate the logger: jest.preset.js 1`] = `
"const nxPreset = require('@nx/jest/preset').default;

module.exports = { ...nxPreset };
"
`;
